### PR TITLE
fix subtitle cutoff

### LINKED ISSUE #13: Subtitles cut off
Subtitles get the last character cut off when rendering. This can be "fixed" by appending spaces to the end of each line in the subtitle file, but should be fixed in code. Likely the space allocated for the strings is short one character.

### PR DESCRIPTION
**Pull request by AshkanArabim**
I fixed a minor bug in `renderer.c` that was causing long single-line subtitles to get cut off. This closes #13.

Here's the behavior before the change:
image

and here's the same subtitle after the change:
image

If you accept this PR, please also add the `hacktoberfest-accepted` tag. :)

### CODE REVIEW

File: renderer.c [hunk 1]
```diff
@@ -167,7 +167,7 @@ void prepSubs(char *str){
 	// Update string to centered and stripped text
 	for (int i=0; i<MAX_CHAR; i++){
 		*(str+i) = ' ';
-		if (i >= offset-1 && i < offset+length-2) *(str+i) = newString[iterator++];
+		if (i >= offset && i < offset+length && newString[iterator] != '\r') *(str+i) = newString[iterator++];
```
Comment (Skote2) [inline renderer.c]: New lines were stripped above. Removal of the carriage return should probably be done in the same place.

**Review by Skote2 — COMMENTED** [review]
Good job, I still need to count this loop out and make sure it's correct but I'll approve after 😅

File: renderer.c (hunk 1 above)
Comment (AshkanArabim) [reply renderer.c]: I'll handle it.

Comment (AshkanArabim) [reply renderer.c]: It should be fine now!

**Comment by aidancrowther** [comment]
Have you validated multiple long subtitles? Such as two very long sentences within a single subtitle block?

**Comment by aidancrowther** [comment]
Primarily, this change would benefit out-of-spec subtitles, as the SRT format sets a 32-character limit on line length. That said, it may still be beneficial to implement some redundancy in order to facilitate easier implementation of other subtitle formats.

The SRT format specification can be found [here](https://loc.gov/preservation/digital/formats//fdd/fdd000569.shtml#:~:text=Structure%20of%20SRT%3A,srt%22%20SRT%20file.)

It may be prudent to enable resizing of the text window should need be for this case, but we will have to investigate whether any mainstream formats exist that allow these long subtitles before investing too much effort into it.

**Comment by AshkanArabim** [comment]
My changes work with this sample SRT file. 

Edit: let me know if you think my test cases are missing something so I can add them!

```
1
00:00:00,010 --> 00:00:02,000
1- edge case

2
00:00:02,010 --> 00:00:04,000
2- quite a long piece of text I expect to get split into two lines as you can see

3
00:00:04,010 --> 00:00:06,000
3- edge
case

4
00:00:06,010 --> 00:00:08,000
4- quite a long piece of text I expect to get split into two lines as you can see
an another piece of very long text that should not even appear as far as I know.

5
00:00:08,010 --> 00:00:10,000
5- line after overflow.

6
00:00:10,010 --> 00:00:12,000
6- generic sub after the last one that overflowed. will it work?


```
